core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1155,7 +1156,8 @@ class DAGScheduler(
       case commitDenied: TaskCommitDenied =>
         // Do nothing here, left up to the TaskScheduler to decide how to handle denied commits
 
-      case ExceptionFailure(className, description, stackTrace, fullStackTrace, metrics) =>
+      case ExceptionFailure(className, description, stackTrace, fullStackTrace, metrics,
+                            exception) =>
```

Comment:
nit: also not something you introduced, but while you're touching this, it can be cleaned up.  If you're not using any of those fields, we can just check the class in the case statement  `case _: ExceptionFailure =>`.  or since there are so many uses of underscore in scala and they can be kinda confusing, you can bind the whole thing to a variable (even though we'll never look at it): `case ef: ExceptionFailure =>`